Controllers/WishListsController.cs

```diff
@@ -19,44 +20,64 @@ namespace Nwazet.Commerce.Controllers {
     public class WishListsController : Controller {
         private readonly IWorkContextAccessor _wca;
         private readonly IWishListServices _wishListServices;
+        private readonly IWishListsUIServices _wishListsUIServices;
         private readonly IContentManager _contentManager;
         private readonly IEnumerable<IProductAttributeExtensionProvider> _attributeExtensionProviders;
         private readonly IShoppingCart _shoppingCart;
         private readonly IWorkflowManager _workflowManager;
         private readonly IEnumerable<IWishListExtensionProvider> _wishListExtensionProviders;
+        private readonly IOrchardServices _orchardServices;
+        private readonly IMembershipService _membershipService;
 
         public WishListsController(
             IWorkContextAccessor wca,
             IWishListServices wishListServices,
+            IWishListsUIServices wishListsUIServices,
             IContentManager contentManager,
             IEnumerable<IProductAttributeExtensionProvider> attributeExtensionProviders,
             IShoppingCart shoppingCart,
             IWorkflowManager workflowManager,
-            IEnumerable<IWishListExtensionProvider> wishListExtensionProviders) {
+            IEnumerable<IWishListExtensionProvider> wishListExtensionProviders,
+            IOrchardServices orchardServices,
+            IMembershipService membershipService) {
 
             _wca = wca;
             _wishListServices = wishListServices;
+            _wishListsUIServices = wishListsUIServices;
             _contentManager = contentManager;
             _attributeExtensionProviders = attributeExtensionProviders;
             _shoppingCart = shoppingCart;
             _workflowManager = workflowManager;
             _wishListExtensionProviders = wishListExtensionProviders;
+            _orchardServices = orchardServices;
+            _membershipService = membershipService;
+
+            T = NullLocalizer.Instance;
         }
 
+        public Localizer T { get; set; }
+
         private const string AttributePrefix = "productattributes.a";
         private const string ExtensionPrefix = "ext.";
 
         [OutputCache(Duration = 0)]
+        public ActionResult Index(string uName, int id = 0) {
+            //the uName string is there to allow accessing other user's wishlists
+            var user = string.IsNullOrWhiteSpace(uName) ? _wca.GetContext().CurrentUser
+                : _membershipService.GetUser(uName);
+
+            WishListListPart selectedList;
+            if (_wishListServices.TryGetWishList(out selectedList, id)) {
+                if (!_orchardServices.Authorizer.Authorize(WishListPermissions.ViewWishLists, selectedList))
+                    return RedirectToAction("Index", new { id = 0 }); //redirect to own wish lists
```

Comment:
Yes.
My assumption is that a user will always be allowed to see their wish lists, and the way the services are implemented, every user will always have at least 1 wish list.
I don't see a reasonable use case where a user would not be allowed to see their own wish lists.
Of course it's possible to implement the services so that users may not have any wishlist. I don't see a case where that would make sense.
Anyway, I can handle that: I am thinking of adding a check and returning an error page when the user has no wishlists they can see. Does that make sense?